### PR TITLE
[benchmark] Add threshold and reorganize post-process for benchmark 

### PR DESCRIPTION
**Pull request by YYStreet**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [benchmark]

*Description:*
 Add threshold and reorganize post-process fo benchmark
 Drop the support for py2 MXNet gpu training and cpu/gpu inference benchmark

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet [hunk 1]
```diff
@@ -49,7 +49,7 @@ echo Preparation Stage took $(( DIFF/60 )) mins to finish>&2
 
 echo Benchmark started: >&2
 START=$(date +%s)
-trigger_test "100m" "run" 2>&1 | tee ${LOG_DIR}/${LOG_FILE}
+trigger_test "60m" "run" 2>&1 | tee ${LOG_DIR}/${LOG_FILE}
```
Comment (YYStreet) [inline test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:52]: Change it to debug under PR context. Will revert it

File: pytorch/buildspec.yml [hunk 2]
```diff
@@ -23,6 +28,16 @@
       example_mnist_file:
         source: docker/build_artifacts/mnist.py
         target: mnist.py
+    inference_context: &INFERENCE_CONTEXT
```
Comment (YYStreet) [inline pytorch/buildspec.yml:31]: Build inference images for testing

File: test/dlc_tests/conftest.py [hunk 3]
```diff
@@ -144,7 +144,7 @@ def ec2_instance(
     }
     extra_volume_size_mapping = [{"DeviceName": "/dev/sda1", "Ebs": {"VolumeSize": 300,}}]
     if (
-        "benchmark" in os.getenv("TEST_TYPE")
+        True
```
Comment (YYStreet) [inline test/dlc_tests/conftest.py:147]: Will convert it back with other config change

File: test/testrunner.py [hunk 4]
```diff
@@ -250,8 +250,10 @@ def main():
     # Do not create EKS cluster for when EIA Only Images are present
     is_all_images_list_eia = all("eia" in image_uri for image_uri in all_image_list)
     eks_cluster_name = None
-    benchmark_mode = "benchmark" in test_type
-    specific_test_type = re.sub("benchmark-", "", test_type) if benchmark_mode else test_type
+    benchmark_mode = True
```
Comment (YYStreet) [inline test/testrunner.py:253]: Will be reverted

File: pytorch/buildspec.yml (hunk 2 above)
Comment (TusharKanekiDey) [reply pytorch/buildspec.yml:31]: So this change will revert in final change

Comment (YYStreet) [reply pytorch/buildspec.yml:31]: yes it will be

File: test/dlc_tests/benchmark/ec2/pytorch/inference/test_performance_pytorch_inference.py [hunk 5]
```diff
@@ -36,7 +38,7 @@ def ec2_performance_pytorch_inference(image_uri, processor, ec2_connection, regi
     commit_info = os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")
     # Run performance inference command, display benchmark results to console
     container_name = f"{repo_name}-performance-{image_tag}-ec2"
-    log_file = f"inference_benchmark_results_{commit_info}_{time_str}.log"
+    log_file = f"synthetic_{commit_info}_{time_str}.log"
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/pytorch/inference/test_performance_pytorch_inference.py:41]: Doesn't using the same log-file name for multiple images cause them to overwrite one another, when the tests are run in parallel?

File: test/dlc_tests/benchmark/ec2/pytorch/inference/test_performance_pytorch_inference.py [hunk 6]
```diff
@@ -49,17 +51,5 @@ def ec2_performance_pytorch_inference(image_uri, processor, ec2_connection, regi
...
-    else:
-        ec2_connection.run(f"cat {log_file} >&2")
-    ec2_connection.run(
-        f"aws s3 cp {log_file} s3://dlinfra-dlc-cicd-performance/pytorch/ec2/inference/{processor}/{python_version}/{log_file}"
-    )
-    ec2_connection.run(
-        f"echo To retrieve complete benchmark log, check s3://dlinfra-dlc-cicd-performance/pytorch/ec2/inference/{processor}/{python_version}/{log_file} >&2"
-    )
+    ec2_performance_upload_result_to_s3_and_validate_performance(ec2_connection, image_uri, log_file, "synthetic",
+                                                                 threshold, post_process_inference, log_file)
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/pytorch/inference/test_performance_pytorch_inference.py:55]: Correct the indentation here.

File: test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py [hunk 7]
```diff
@@ -37,15 +43,41 @@ def execute_pytorch_gpu_py3_imagenet_ec2_training_performance_test(connection, e
 
     # Make sure we are logged into ECR so we can pull the image
     connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
-
     connection.run(f"nvidia-docker pull -q {ecr_uri}")
-
+    timestamp = time.strftime('%Y-%m-%d-%H-%M-%S')
+    log_name = f"imagenet_{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}_{timestamp}.txt"
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py:48]: Doesn't using this as the log name cause conflicts between multiple tests running in parallel, and make them overwrite the data the other produces in this file?

File: test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py [hunk 8]
```diff
@@ -37,15 +43,41 @@ def execute_pytorch_gpu_py3_imagenet_ec2_training_performance_test(connection, e
...
+    log_location = os.path.join(container_test_local_dir, "benchmark", "logs", log_name)
     # Run training command, display benchmark results to console
     try:
         connection.run(
-            f"nvidia-docker run --user root -e COMMIT_INFO={os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')} --shm-size 8G --env OMP_NUM_THREADS=1 --name {container_name} "
+            f"nvidia-docker run --user root -e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} "
+            f"--shm-size 8G --env OMP_NUM_THREADS=1 --name {container_name} "
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py:54]: Thanks.

nit: Could you also move `-e LOGFILE=...` to its own line?

File: test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py [hunk 9]
```diff
@@ -37,15 +43,41 @@ def execute_pytorch_gpu_py3_imagenet_ec2_training_performance_test(connection, e
...
             f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} "
             f"-v /home/ubuntu/:/root/:delegated "
             f"{ecr_uri} {os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
     finally:
         connection.run(f"docker rm -f {container_name}", warn=True, hide=True)
+    ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location, "imagenet",
+                                                                 {"Cost": PYTORCH_TRAINING_GPU_IMAGENET_THRESHOLD},
+                                                                 post_process_pytorch_gpu_py3_imagenet_ec2_training_performance,
+                                                                 log_name)
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/pytorch/training/test_performance_pytorch_training.py:63]: Indentation

File: test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py [hunk 10]
```diff
@@ -1,27 +1,34 @@
 import os
 import time
+import re
 import pytest
+from src.benchmark_metrics import TENSORFLOW2_INFERENCE_GPU_THRESHOLD, TENSORFLOW2_INFERENCE_CPU_THRESHOLD, \
+    TENSORFLOW1_INFERENCE_GPU_THRESHOLD, TENSORFLOW1_INFERENCE_CPU_THRESHOLD
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py:6]: Fix according to recommendation by `black`, where max line-length is 120.

File: test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py [hunk 11]
```diff
@@ -1,27 +1,34 @@
...
 
 
-def ec2_performance_tensorflow_inference(image_uri, processor, ec2_connection, region):
+def ec2_performance_tensorflow_inference(image_uri, processor, ec2_connection, region, threshold):
     docker_cmd = "nvidia-docker" if processor == "gpu" else "docker"
-    python_version = "py2" if "py2" in image_uri else "py3"
     container_test_local_dir = os.path.join("$HOME", "container_tests")
     tf_version = "1" if is_tf1(image_uri) else "2"
-    tf_api_version = '1.15' if tf_version == '1' else '2.2.0'
+    tf_api_version = '1.15' if tf_version == '1' else '2.3.0'
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py:31]: Do not hard-code this. Please obtain this information from the `image_uri`.

File: test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py [hunk 12]
```diff
@@ -35,16 +42,12 @@ def ec2_performance_tensorflow_inference(image_uri, processor, ec2_connection, r
     )
     time_str = time.strftime('%Y-%m-%d-%H-%M-%S')
     commit_info = os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")
-    log_file = f"inference_benchmark_results_{commit_info}_{time_str}.log"
+    log_file = f"synthetic_{commit_info}_{time_str}.log"
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py:45]: Will naming this log file the same for all images being tested in parallel cause the scripts to overwrite one another's logs in the same file (if time_str doesn't change between two parallel tests)?

File: test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py [hunk 13]
```diff
@@ -35,16 +42,12 @@ def ec2_performance_tensorflow_inference(image_uri, processor, ec2_connection, r
...
     )
-    ec2_connection.run(
-        f"echo Benchmark Results: >&2;"
-        f"echo Tensorflow{tf_version} Inference {processor} {python_version} >&2"
-    )
-    ec2_connection.run(f"tail {log_file} >&2")
-    ec2_connection.run(f"aws s3 cp {log_file} {BENCHMARK_RESULTS_S3_BUCKET}/tensorflow{tf_version}/ec2/inference/{processor}/{python_version}/{log_file}")
-    ec2_connection.run(f"echo To retrieve complete benchmark log, check s3://dlinfra-dlc-cicd-performance/tensorflow{tf_version}/ec2/inference/{processor}/{python_version}/{log_file} >&2")
+    ec2_performance_upload_result_to_s3_and_validate_performance(ec2_connection, image_uri, log_file, "synthetic",
+                                                                 threshold, post_process_inference, log_file)
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/tensorflow/inference/test_performance_tensorflow_inference.py:52]: Indentation

File: test/dlc_tests/benchmark/ec2/tensorflow/training/test_performance_tensorflow_training.py [hunk 14]
```diff
@@ -1,10 +1,13 @@
...
 import pytest
 
-from test.test_utils import CONTAINER_TESTS_PREFIX
+from test.test_utils import CONTAINER_TESTS_PREFIX, is_tf2
 from test.test_utils.ec2 import execute_ec2_training_performance_test
-
+from src.benchmark_metrics import TENSORFLOW2_TRAINING_CPU_SYNTHETIC_THRESHOLD, \
+    TENSORFLOW2_TRAINING_GPU_SYNTHETIC_THRESHOLD, TENSORFLOW2_TRAINING_GPU_IMAGENET_THRESHOLD, \
+    TENSORFLOW1_TRAINING_CPU_SYNTHETIC_THRESHOLD, \
+    TENSORFLOW1_TRAINING_GPU_SYNTHETIC_THRESHOLD, TENSORFLOW1_TRAINING_GPU_IMAGENET_THRESHOLD
```
Comment (saimidu) [inline test/dlc_tests/benchmark/ec2/tensorflow/training/test_performance_tensorflow_training.py:10]: Fix indentation according to `black`.

File: test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_performance_tensorflow_sm_training.py [hunk 15]
```diff
@@ -76,25 +79,42 @@ def test_tensorflow_sagemaker_training_performance(tensorflow_training, num_node
...
+    throughput /= num_nodes
 
     assert run_out.ok, (f"Benchmark Test failed with return code {run_out.return_code}. "
                         f"Test results can be found at {os.path.join(target_upload_location, log_file)}")
 
+    threshold = (TENSORFLOW2_SM_TRAINING_CPU_1NODE_THRESHOLD if num_nodes == 1
+                 else TENSORFLOW2_SM_TRAINING_CPU_4NODE_THRESHOLD) \
+        if processor == "cpu" \
+        else TENSORFLOW2_SM_TRAINING_GPU_1NODE_THRESHOLD if num_nodes == 1 \
+        else TENSORFLOW2_SM_TRAINING_GPU_4NODE_THRESHOLD
```
Comment (saimidu) [inline test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_performance_tensorflow_sm_training.py:92]: Fix indentation.

File: test/dlc_tests/container_tests/bin/benchmark/run_mxnet_training_performance_gpu [hunk 16]
```diff
@@ -22,21 +20,15 @@ echo Fetch finished >&2
...
-tail ${LOG_DIR}/"${LOG_FILE}" >&2 # Display only the results to console
-aws s3 cp ${LOG_DIR}/"${LOG_FILE}" s3://dlinfra-dlc-cicd-performance/mxnet/ec2/training/gpu/py"${PYTHON_VERSION}"/"${LOG_FILE}"
-echo To retrieve complete benchmark log, check s3://dlinfra-dlc-cicd-performance/mxnet/ec2/training/gpu/py"${PYTHON_VERSION}"/"${LOG_FILE}" >&2
+echo Script took $(( DIFF/60 )) mins to finish >> "${LOG_FILE}"
 set +e
 
 rm -rf ${HOME_DIR}/artifacts/mxnet/deep-learning-benchmark-mirror
 
 exit 0
+#TODO: line 23: prepend '--num-epochs 20'
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/benchmark/run_mxnet_training_performance_gpu:34]: Is this something that needs to be done at the end of the PR review?

File: test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet [hunk 17]
```diff
@@ -49,15 +47,9 @@ echo Preparation Stage took $(( DIFF/60 )) mins to finish>&2
 
 echo Benchmark started: >&2
 START=$(date +%s)
-trigger_test "100m" "run" 2>&1 | tee ${LOG_DIR}/${LOG_FILE}
-python /test/bin/benchmark/get_pytorch_performance_num.py ${LOG_DIR}/"${LOG_FILE}" >> ${LOG_DIR}/"${LOG_FILE}"
+trigger_test "60m" "run" 2>&1 | tee "${LOG_FILE}" #TODO: need reverted to 100m
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/benchmark/run_pytorch_training_performance_gpu_imagenet:50]: Revert at the end of PR review.

File: test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_training_performance_gpu_imagenet [hunk 18]
```diff
@@ -20,15 +17,11 @@ mkdir -p ${LOG_DIR}
...
 START=$(date +%s)
-timeout 100m python scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --batch_size=256 --model=resnet50_v1.5 --optimizer=momentum --variable_update=replicated --nodistortions --gradient_repacking=2 --num_gpus=8 --num_epochs=91 --weight_decay=1e-4 --use_fp16 --all_reduce_spec=nccl --save_summaries_steps=0 --summary_verbosity=1 --num_warmup_batches=0 --train_dir=$HOME/test00 --compute_lr_on_cpu=True --single_l2_loss_op=True --loss_type_to_report=base_loss --data_name=imagenet 2>&1 | tee ${LOG_DIR}/${LOG_FILE}
+timeout 20m python scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --batch_size=256 --model=resnet50_v1.5 --optimizer=momentum --variable_update=replicated --nodistortions --gradient_repacking=2 --num_gpus=8 --num_epochs=91 --weight_decay=1e-4 --use_fp16 --all_reduce_spec=nccl --save_summaries_steps=0 --summary_verbosity=1 --num_warmup_batches=0 --train_dir=$HOME/test00 --compute_lr_on_cpu=True --single_l2_loss_op=True --loss_type_to_report=base_loss --data_name=imagenet 2>&1 | tee "${LOG_FILE}"
+#TODO:revert to 100m
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_training_performance_gpu_imagenet:21]: Revert at the end of PR review.

File: test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_training_performance_gpu_synthetic [hunk 19]
```diff
@@ -20,15 +17,11 @@ mkdir -p ${LOG_DIR}
 set -e
 git clone https://github.com/tensorflow/benchmarks.git ${HOME_DIR}/artifacts/tensorflow/benchmarks && cd ${HOME_DIR}/artifacts/tensorflow/benchmarks && git checkout ${GIT_BRANCH}
 START=$(date +%s)
-timeout 100m python scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --batch_size=256 --model=resnet50_v1.5 --optimizer=momentum --variable_update=replicated --nodistortions --gradient_repacking=2 --num_gpus=8 --num_epochs=91 --weight_decay=1e-4 --use_fp16 --all_reduce_spec=nccl --save_summaries_steps=0 --summary_verbosity=1 --num_warmup_batches=0 --train_dir=$HOME/test00 --compute_lr_on_cpu=True --single_l2_loss_op=True --loss_type_to_report=base_loss 2>&1 | tee ${LOG_DIR}/${LOG_FILE}
+timeout 20m python scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --batch_size=256 --model=resnet50_v1.5 --optimizer=momentum --variable_update=replicated --nodistortions --gradient_repacking=2 --num_gpus=8 --num_epochs=91 --weight_decay=1e-4 --use_fp16 --all_reduce_spec=nccl --save_summaries_steps=0 --summary_verbosity=1 --num_warmup_batches=0 --train_dir=$HOME/test00 --compute_lr_on_cpu=True --single_l2_loss_op=True --loss_type_to_report=base_loss 2>&1 | tee "${LOG_FILE}"
+#TODO:revert to 100m
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/benchmark/run_tensorflow_training_performance_gpu_synthetic:21]: Revert at the end of PR review.

File: test/test_utils/ec2.py [hunk 20]
```diff
@@ -64,8 +66,8 @@ def get_ec2_accelerator_type(default, processor):
 
 
 def launch_instance(
-    ami_id, instance_type, ei_accelerator_type, ec2_key_name=None, region=DEFAULT_REGION, user_data=None,
-    iam_instance_profile_name=None, instance_name=""
+        ami_id, instance_type, ei_accelerator_type, ec2_key_name=None, region=DEFAULT_REGION, user_data=None,
+        iam_instance_profile_name=None, instance_name=""
```
Comment (saimidu) [inline test/test_utils/ec2.py:70]: Why the indent here?

File: test/test_utils/ec2.py [hunk 21]
```diff
@@ -387,8 +389,8 @@ def get_ec2_fabric_connection(instance_id, instance_pem_file, region):
     return conn
 
 
-
-def execute_ec2_training_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION, executable="bash", large_shm=False, host_network=False,
+def execute_ec2_training_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION, executable="bash", large_shm=False,
+                              host_network=False,
                               container_name='ec2_training_container'):
```
Comment (saimidu) [inline test/test_utils/ec2.py:394]: Please correct the indentation here (as well as other function definitions.

File: test/test_utils/ec2.py [hunk 22]
```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
...
 
 
-def execute_ec2_training_performance_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION):
+def execute_ec2_training_performance_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION,
+                                          post_process=None, data_source="", threshold=None):
     docker_cmd = "nvidia-docker" if "gpu" in ecr_uri else "docker"
     container_test_local_dir = os.path.join("$HOME", "container_tests")
 
+    timestamp = time.strftime('%Y-%m-%d-%H-%M-%S')
+    log_name = f"{data_source}_results_{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}_{timestamp}.txt"
```
Comment (saimidu) [inline test/test_utils/ec2.py:446]: Please ensure there are no conflicts in log file names caused by multiple tests running in parallel. Rename the file to something much more specific, if needed.

File: test/test_utils/ec2.py [hunk 23]
```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
...
 
 
-def execute_ec2_inference_performance_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION):
+def execute_ec2_inference_performance_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION,
+                                           post_process=None, data_source="", threshold=None):
     docker_cmd = "nvidia-docker" if "gpu" in ecr_uri else "docker"
     container_test_local_dir = os.path.join("$HOME", "container_tests")
-
+    timestamp = time.strftime('%Y-%m-%d-%H-%M-%S')
+    log_name = f"{data_source}_results_{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}_{timestamp}.txt"
```
Comment (saimidu) [inline test/test_utils/ec2.py:470]: Please ensure there are no conflicts in log file names caused by multiple tests running in parallel. Rename the file to something much more specific, if needed.

File: test/test_utils/ec2.py [hunk 24]
```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
...
     connection.run(
         f"{docker_cmd} run -d --name {container_name} "
-        f"-e COMMIT_INFO={os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')} "
+        f"-e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} "
         f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} {ecr_uri}"
     )
     try:
-        connection.run(f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
+        connection.run(
+            f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
```
Comment (saimidu) [inline test/test_utils/ec2.py:485]: nit: Move the closing-parenthesis at the end of this line to the next line. Example:
```
connection.run(
    f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}"
)
```

File: test/test_utils/ec2.py [hunk 25]
```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
...
         connection.run(f"docker rm -f {container_name}")
+    log_location = os.path.join(container_test_local_dir, "benchmark", "logs", log_name)
+    ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location, data_source,
+                                                                 threshold, post_process, log_name)
+
+
+def ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location,
+                                                                 data_source, threshold, post_process, log_name):
+    framework = "tensorflow" if "tensorflow" in ecr_uri else "mxnet" if "mxnet" in ecr_uri else "pytorch"
+    framework_version = re.search(r"[0-9]+(\.\d+){2}", ecr_uri).group()
```
Comment (saimidu) [inline test/test_utils/ec2.py:498]: `r"\d"` is exactly the same as `r"[0-9]"`. Please use one or the other.

File: test/test_utils/ec2.py [hunk 26]
```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
...
+        performance_statement = f'{framework} {framework_version} ec2 {work_type} {processor} {py_version} ' \
+                                f'{data_source} {k} {description}: {v} {unit}, threshold: {threshold[k]} {unit}'
+        connection.run(
+            f"echo {performance_statement} | sudo tee -a {log_location}")
+        LOGGER.info(
+            f"{performance_statement}")
+    connection.run(
+        f"aws s3 cp {log_location} {s3_location}")
+    connection.run(
+        f"echo To retrieve complete benchmark log, check {s3_location} >&2")
```
Comment (saimidu) [inline test/test_utils/ec2.py:524]: Fix the closing-parenthesis in all these multi-line function calls.

File: test/test_utils/ec2.py (hunk 22 above)
Comment (YYStreet) [reply test/test_utils/ec2.py:446]: ack. should have no conflicts

File: test/test_utils/ec2.py (hunk 23 above)
Comment (YYStreet) [reply test/test_utils/ec2.py:470]: ack

File: test/test_utils/ec2.py (hunk 24 above)
Comment (YYStreet) [reply test/test_utils/ec2.py:485]: will do

File: test/test_utils/ec2.py [hunk 27]
```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
...
         raise Exception("Failed to exec benchmark command.\n", e)
     finally:
         connection.run(f"docker rm -f {container_name}")
+    log_location = os.path.join(container_test_local_dir, "benchmark", "logs", log_name)
+    ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location, data_source,
+                                                                 threshold, post_process, log_name)
+
+
+def ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location,
+                                                                 data_source, threshold, post_process, log_name):
```
[the remark this answers is not in the document]
Comment (YYStreet) [reply test/test_utils/ec2.py:496]: Will use a shorter function name instead

File: test/test_utils/ec2.py (hunk 25 above)
Comment (YYStreet) [reply test/test_utils/ec2.py:498]: '\d' is more concise. will turn to use this one

File: test/test_utils/ec2.py (hunk 26 above)
Comment (YYStreet) [reply test/test_utils/ec2.py:524]: Will have them in separate lines

**Pull request merged**